### ISSUE TITLE
Durable Task Core: JsonDataConverter throws exception on handling the loop referenced object

### ISSUE THREAD

**Issue by jundayin** [Azure/durabletask#863]
Recently we discover that DTF throws exception when the output of the TaskActivity contains circular referenced object. Usually it is generated by EF Core and we would like to hear owner of DTF and see if this is a good practice to contain large object in payload.

The current implementation of the JsonDataConverter does not include the ReferenceLoopHandling setting and it throws upon that. 
https://github.com/Azure/durabletask/blob/58c07a36d443dcd82ab0dd60cf71ac3cdcc97c7f/src/DurableTask.Core/Serializing/JsonDataConverter.cs#L37-L48

Is this by design or it is a miss? I will include a minimum repro later

**Comment by cgillum** [comment]
It's not a good practice to put large objects in the return payload because the cost of deserialization is high and must be paid multiple times, once for each orchestration replay. You really should try to avoid it if you can.

There are some workarounds that allow you to use your own serializer, such as deriving directly from `TaskActivity` (instead of `TaskActivity<T1, T2>`) though they may not be particularly simple.

**Comment by jundayin** [comment]
Thanks for the explanation.

Can I conclude that it is by design that there is no such ReferenceLoopHandling in the default serializer considering other data store providers usage? If not, I think I can go ahead to create an PR to fix it

**Comment by cgillum** [comment]
I don't believe it's by design. I think the original maintainers simply used the default settings and didn't have a specific reason to change them.
